### ISSUE TITLE
Bug: COPY with a trailing backslash yields an error

### ISSUE THREAD

**Issue by thiagowfx** [hadolint/hadolint#1060]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

A `Dockerfile` with this snippet:

```
[...]
COPY --from=buildstage /build/go/REDACTED/docker-build/cloudformation-expired.json\
  /root/cloudformation-expired.json
[...]
```

should be properly parsed by `hadolint`.

### Actual behavior

```
% hadolint Dockerfile
Dockerfile:40:36 unexpected end of line. At least two arguments are required for COPY
```

It can be obviously fixed by removing the trailing backslash and thus joining the two lines.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
% hadolint --version
Haskell Dockerfile Linter 2.12.0
```
(from homebrew)

**Dockerfile** (if relevant)

```Dockerfile
ARG CODE_COVERAGE=false

FROM golang:1.23-alpine3.20 AS build-prep

RUN apk add gcc musl-dev

RUN go install github.com/jstemmer/go-junit-report@latest

COPY . /build/go

WORKDIR /build/go/REDACTEDweb

FROM build-prep AS buildstage-code-coverage-true

RUN --mount=type=cache,target=/go/pkg/mod \
  --mount=type=cache,target=/root/.cache/go-build \
  go build -cover -covermode=atomic -o app

FROM build-prep AS buildstage-code-coverage-false

RUN --mount=type=cache,target=/go/pkg/mod \
  --mount=type=cache,target=/root/.cache/go-build \
  go build -o app

FROM buildstage-code-coverage-${CODE_COVERAGE} AS buildstage

FROM alpine:3.18 AS runtime-prep

RUN apk add --no-cache ca-certificates tzdata curl
COPY --from=buildstage /build/go/REDACTEDweb/docker-build/REDACTEDEnv.yaml /root/REDACTEDEnv.yaml
COPY --from=buildstage /build/go/REDACTEDweb/app /bin/REDACTEDweb
COPY --from=buildstage /build/go/REDACTEDweb/docker-build/cloudformation-expired.json\
  /root/cloudformation-expired.json
```

**Additional environment details (OS, stack version, etc.)**

This bug was tricky to track down, because https://github.com/hadolint/hadolint-action does not display the error in the log output. I could only see the error when running `hadolint` directly against the affected `Dockerfile`.

To reproduce: Run `hadolint Dockerfile` with the contents above.
